lib/service_extensions.dart

```diff
@@ -6,15 +6,17 @@ library service_extensions;
 
 import 'ui/icons.dart';
 
-class ServiceExtensionDescription<T> {
-  const ServiceExtensionDescription({
+class ToggleableServiceExtensionDescription<T> {
+  ToggleableServiceExtensionDescription({
```

Comment:
nit: a nice strategy for cases like this where constructors have surprising side effects is make sure nobody outside this library call call this.
Make the constructor private:
ToggleableServiceExtensionDescription._
now you can be confident that people are only calling it from this file which is a good thing as registering extensions after the fact would cause bugs.

Reply:
Done.